google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java

```diff
@@ -0,0 +1,312 @@
+    String jvmName = ManagementFactory.getRuntimeMXBean().getName();
+    if (jvmName.indexOf(64) < 1) {
+      String hostname = "localhost";
+
+      try {
+        hostname = InetAddress.getLocalHost().getHostName();
+      } catch (UnknownHostException e) {
+        logger.log(Level.INFO, "Unable to get the hostname.", e);
+      }
+
+      return "java-" + (new SecureRandom()).nextInt() + "@" + hostname;
+    } else {
+      return "java-" + jvmName;
+    }
+  }
+
+  @VisibleForTesting
+  static MetricKind createMetricKind(Type type) {
+    switch (type) {
+      case GAUGE_INT64:
+      case GAUGE_DOUBLE:
+        return MetricKind.GAUGE;
+      case CUMULATIVE_DOUBLE:
+      case CUMULATIVE_INT64:
+      case CUMULATIVE_DISTRIBUTION:
+        return MetricKind.CUMULATIVE;
+      default:
+        return MetricKind.UNRECOGNIZED;
+    }
+  }
+
+  @VisibleForTesting
+  static com.google.api.MetricDescriptor.ValueType createValueType(Type type) {
+    switch (type) {
+      case GAUGE_DOUBLE:
+      case CUMULATIVE_DOUBLE:
+        return com.google.api.MetricDescriptor.ValueType.DOUBLE;
+      case GAUGE_INT64:
+      case CUMULATIVE_INT64:
+        return com.google.api.MetricDescriptor.ValueType.INT64;
+      case GAUGE_DISTRIBUTION:
+      case CUMULATIVE_DISTRIBUTION:
+        return com.google.api.MetricDescriptor.ValueType.DISTRIBUTION;
+      default:
+        return com.google.api.MetricDescriptor.ValueType.UNRECOGNIZED;
+    }
+  }
+
+  static TimeSeries convertTimeSeries(
+      String metricName,
+      io.opencensus.metrics.export.MetricDescriptor.Type metricType,
+      List<LabelKey> labelKeys,
+      io.opencensus.metrics.export.TimeSeries timeSeries,
+      String clientId,
+      MonitoredResource monitoredResource) {
+
+    MonitoredResource.Builder monitoredResourceBuilder = monitoredResource.toBuilder();
+
+    List<LabelKey> metricTagKeys = new ArrayList<>();
+    List<LabelValue> metricTagValues = new ArrayList<>();
+
+    List<LabelValue> labelValues = timeSeries.getLabelValues();
+    for (int i = 0; i < labelValues.size(); i++) {
+      if (PROMOTED_RESOURCE_LABELS.contains(labelKeys.get(i).getKey())) {
+        monitoredResourceBuilder.putLabels(
+            labelKeys.get(i).getKey(), labelValues.get(i).getValue());
+      } else {
+        metricTagKeys.add(labelKeys.get(i));
+        metricTagValues.add(labelValues.get(i));
+      }
+    }
+    metricTagKeys.add(LabelKey.create(BuiltinMeasureConstants.CLIENT_UID.getName(), "client id"));
+    metricTagValues.add(LabelValue.create(clientId));
+
+    TimeSeries.Builder builder = TimeSeries.newBuilder();
+    builder.setMetricKind(createMetricKind(metricType));
+    builder.setResource(monitoredResourceBuilder.build());
+    builder.setValueType(createValueType(metricType));
+    builder.setMetric(createMetric(metricName, metricTagKeys, metricTagValues));
+    io.opencensus.common.Timestamp startTimeStamp = timeSeries.getStartTimestamp();
```

Comment:
why long import? (everywhere else as well)

Reply:
Also can you order these in a logical way? ie put metric & metrickind together. And order the setters in a hierarchical order: resource, metric, value